docs/docsite/rst/user_guide/playbook_pathing.rst

```diff
@@ -23,15 +23,20 @@ Relative paths used in a task typically refer to remote files and directories on
 Resolving local paths
 ---------------------
 
-Ansible tries to find local files first in the current task's role, then in other roles that included or depend on the current role, and finally in the current play. This way, if multiple files with the same filename exist, Ansible will find the file that is closest to the current task and that is most likely to be file you wanted.
+Ansible tries to find local files first relative to the current task's role, then relative to other roles that included or depend on the current role, then relative to the file in which the task is defined, and finally relative to the current play. This way, if multiple files with the same filename exist, Ansible will find the file that is closest to the current task and that is most likely to be file you wanted.
 
-Specifically, Ansible searches these locations:
+Specifically, Ansible tries to find the file
 
-1. In the active role's matching "files", "vars", "templates", or "tasks" subdirectory, depending on which kind of file it is searching for.
-2. In a previous role's matching subdirectory, if that role called into this current role with an `include_role` or `import_role` task or with a role dependency.
-3. In the active play's matching subdirectory as well as directly in the active play's directory.
+1. In the active role's "files", "vars", "templates", or "tasks" subdirectory, depending on which kind of file it is searching for.
```

Comment:
tasks always gets searched for the other types, why they were separate.

Reply:
Oh so it's
1. in the subdirectories
2. in the directory itself
3. in subdirectories under tasks/
4. in tasks/
for each level? I think I'll have to re-phrase the whole thing then, that's a bit much to put into that list.

Reply:
3 & 4 are really 'one' step, also 'tasks/' can also be #1 if the 'searcher' is include_tasks/import_tasks

Reply:
so first is  role_path/subdir (where subdir can be vars/files/template/tasks depending on 'caller)
2nd is role_path
3rd is 'dir of task itself', which 'normally' is role_path/tasks/ but can be a subdir of it
then it goes to parent

Reply:
Thanks for the help! My step 2 still looks a bit complex but I think it's much better now with your suggestions. As I mentioned in the other thread, for me this step

> 3rd is 'dir of task itself', which 'normally' is role_path/tasks/ but can be a subdir of it or 'role_path/handlers' in the case of a handler

is swapped with

> then it goes to parent

. Ansible only searches that directory after it failed to find the file anywhere in the role hierarchy.

I'm closing the other one, this seems to be the same issue and there's no point discussing this in two places.